GitCommands/GitCommands.csproj

```diff
@@ -59,6 +60,7 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <Prefer32Bit>false</Prefer32Bit>
     <CodeAnalysisRuleSet>..\GitExtensions.ruleset</CodeAnalysisRuleSet>
+    <DocumentationFile>bin\Release\GitCommands.xml</DocumentationFile>
```

Comment:
Every project now generates a documentation file for all build configurations (i.e. Debug and Release).

Reply:
:bulb: We can move this to **Directory.Build.props** and remove this pair of lines from every project.

⚠️ This may interfere with submodule projects, ~~so if you like the idea it's probably best to do it in a separate pull request after this one is merged.~~ (See end of this comment)


```xml
<PropertyGroup>
  <DocumentationFile>$(OutputPath)$(AssemblyName).xml</DocumentationFile>
</PropertyGroup>
```

**Edit:** On second thought, given the number of files this would make simpler from this review, it may be worth examining. Since the submodules are in **Externals**, *if you run into problems with the approach* you could create a file **Externals\Directory.Build.props** which overrides this setting just for the submodules.

Reply:
@sharwell I've been trying to get this to work but it seems that `AssemblyName` is an empty string at this point, as we just end up with a file called `.xml`. Seems like we need `$(Configuration)` in there too to get the file into the `bin/.../` folder.

```xml
<Target Name="Foo" BeforeTargets="Build">
  <Message Text="$(OutputPath)\$(Configuration)$(AssemblyName).xml" Importance="high"/>
</Target>
```

This outputs `bin\Debug\Debug\GitExtUtils.xml` (for example) so perhaps the `Directory.Build.props` evaluation happens before the `AssemblyName` property has been set.

Do you know a way around this?

Reply:
> Do you know a way around this?

Instead of defining `DocumentationFile`, use this:

```
<GenerateDocumentationFile>true</GenerateDocumentationFile>
```

The default value it applies will match the one you wanted.

Reply:
(It's possible the previous comment won't work, in which case we'll just go with the original approach and I can experiment with alternatives separately.)

Reply:
Thanks for following up. I tried `GenerateDocumentationFile` and it didn't generate anything. Looks like that option [might be for dotnet SDK only](https://github.com/dotnet/sdk/issues/67).

I can't find an easier approach here. Will apply your other feedback to the PR and force push. If it's merged without squashing, then you could easily enough revert that one commit if you find a better approach that I missed.